lib/sycamore/sycamore/utils/time_trace.py

```diff
@@ -4,8 +4,59 @@
 import resource
 import threading
 import functools
+import logging
 from sys import platform
 
+zero_time = time.time_ns()
+# Mac reports ru_maxrss in bytes. Linux reports it in kB.
+if platform == "darwin":
+    RSS_MULTIPLIER = 1
+else:
+    RSS_MULTIPLIER = 1024
+
+
+class TimeTraceData:
+    def __init__(self, t0, t1, user, sys, rss):
+        self.t0 = t0  # ns
+        self.t1 = t1  # ns
+        self.user = user  # ns
+        self.sys = sys  # ns
+        self.rss = rss  # bytes
+
+    def wall_s(self):
+        return (self.t1 - self.t0) / 1.0e9
+
+    def user_s(self):
+        return self.user / 1.0e9
+
+    def sys_s(self):
+        return self.sys / 1.0e9
+
+    def rss_mib(self):
```

Comment:
I know why people use MiB instead of MB, but still whenever I see mib I think of SNMP.